### ISSUE TITLE
I can't start SD in A1111 notebook ("None" in local URL)

### ISSUE THREAD

**Issue by elo0i** [TheLastBen/fast-stable-diffusion#2276]
![Captura de pantalla 2023-06-26 103104](https://github.com/TheLastBen/fast-stable-diffusion/assets/114620017/901051b6-0277-40d0-8943-f05c8a3014a8)

I'm getting this error (screenshot above) when executing this cell as I always did (I'm using vast.ai, on runpod works fine)

Start Stable-Diffusion[¶](https://81.166.162.13:10951/lab/tree/workspace/RNPD-A1111.ipynb#Start-Stable-Diffusion)

User = ""
​
Password= ""
​
# Add credentials to your Gradio interface (optional).
​
#-----------------
configf=sd(User, Password, model) if 'model' in locals() else sd(User, Password, "");import gradio;gradio.close_all()
!python /workspace/sd/stable-diffusion-webui/webui.py $configf
2023-06-26 08:27:35,209 - ControlNet - INFO - ControlNet v1.1.227
ControlNet preprocessor location: /workspace/sd/stable-diffusion-webui/extensions/sd-webui-controlnet/annotator/downloads
2023-06-26 08:27:35,283 - ControlNet - INFO - ControlNet v1.1.227
Loading weights [4c86efd062] from /workspace/sd/stable-diffusion-webui/models/Stable-diffusion/SDv1-5.ckpt
Running on local URL: https://none-3000.proxy.runpod.net/
✔ Connected
Startup time: 5.1s (import torch: 1.5s, import gradio: 0.7s, import ldm: 0.2s, other imports: 1.5s, load scripts: 0.6s, create ui: 0.3s, gradio launch: 0.2s).
Creating model from config: /workspace/sd/stable-diffusion-webui/configs/v1-inference.yaml
LatentDiffusion: Running in eps-prediction mode
DiffusionWrapper has 859.52 M params.
Applying optimization: sdp... done.
Textual inversion embeddings loaded(0): 
Model loaded in 3.8s (load weights from disk: 1.3s, load config: 0.2s, create model: 0.5s, apply weights to model: 0.7s, apply half(): 0.7s, move model to device: 0.4s).

**Comment by TheLastBen** [comment]
the notebooks are designed for runpod, they are not available on vast.ai

**Comment by TheLastBen** [comment]
do you want to use it as an API ?

**Comment by elo0i** [comment]
> do you want to use it as an API ?
Yes!! (I don't know if I have to use --api or --nowebui but I use --api). I have my script working on runpod and works very well but I need to do it in vast.ai given the limitations of runpod API. I just need to run SD in some way on the instance and then generate images via commands in the script (the best solution I found is launch with --api)

**Comment by TheLastBen** [comment]
there is a runpod API based on this repo's training method 
https://docs.runpod.io/reference/dreambooth-sd-v15

**Comment by elo0i** [comment]
> there is a runpod API based on this repo's training method https://docs.runpod.io/reference/dreambooth-sd-v15

Yes I know but that's not the problem, I already have my own Dreambooth Training API running on vast.ai and it works well, the problem is I need a INFERENCE "script" to generate x images with x prompt. I dont even need API "mode" but the thing is I want to do it with the custom trained model of each user of my web, I already know how to manage that but I am not being able to run inference on vast.ai with this repo (I would love to do it with your repo since it is the one that gives me better results). My doubt is, can I create a instance on vast.ai with your docker image (fast-sd:2.1.0), then execute a python script that generates x images with the prompt specified in the script (I don't need api or webui but launch with --api is the best solution i found)

**Comment by elo0i** [comment]
Oh and another doubt. I tried adapting the runpod one to vast.ai, would be easier to adapt the colab pro notebook to work on jupyter in vast.ai? Maybe I am saying something stupid but on Colab the URL generated is fully gradio and has nothing to do with runpod right?

**Comment by elo0i** [comment]
Or someone knows any other repo to do what I say above?
